backend/apps/monkvision/apis/lib/nlp.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * NLP Model and Predictions
+ *  
+ * (C) 2022 TekMonks. All rights reserved.
+ * License: See enclosed license.txt file.
+ */
+
+const fs = require("fs");
+const Mustache = require("mustache");
+const utils = require(`${APP_CONSTANTS.LIB_DIR}/nlp/utils.js`);
+const tf = require("@tensorflow/tfjs");
+const use = require("@tensorflow-models/universal-sentence-encoder");
+require('@tensorflow/tfjs-node');
+
+// Load the englishModel
+const englishModel = require(`${APP_CONSTANTS.CONF_DIR}/englishModel.json`);
+
+// Global declarations
+const MODELS = {}, TENSORS = {}, INTENTS = {}, METRICS = Object.assign({}, englishModel.metrics), 
+    DURATIONS = Object.assign({}, englishModel.durations), RESOURCES = Object.assign({}, englishModel.resources);
+
+// Train Model (JSON)
+INTENTS["metrics"] = JSON.parse(Mustache.render(fs.readFileSync(`${APP_CONSTANTS.CONF_DIR}/models/metric_intents.json`, "utf8"), englishModel));
+INTENTS["durations"] = JSON.parse(Mustache.render(fs.readFileSync(`${APP_CONSTANTS.CONF_DIR}/models/duration_intents.json`, "utf8"), englishModel));
+INTENTS["resources"] = JSON.parse(Mustache.render(fs.readFileSync(`${APP_CONSTANTS.CONF_DIR}/models/resource_intents.json`, "utf8"), englishModel));
+
+exports.init = async () => {
```

Comment:
This should be initSync and INTENTS which use Sync methods should be initialized here not in the freeflow header of the code!

Reply:
nls.js already never uses any async code in it's init. And should actually. doit (reading intents). I don't see why it is even named `init` in the first place.